src/test/java/io/jenkins/plugins/gcr/GithubClientTests.java

```diff
@@ -57,10 +57,8 @@ private GithubPayload createMockPayload() {
 
     private PluginEnvironment createMockEnvironment() {
         EnvVars envVars = new EnvVars(
-                "ghprbPullId", "34",
```

Comment:
Can you elaborate the tests to use the two different sets of environment variables that are now supported? This will give the best level of coverage.

Reply:
give me a couple of hours, I will create 2 units tests, one for each case.

Reply:
added.
If you are happy with that, I think the next step for me, is to compile it again, test it on my environment, and maybe you can do the same on your side?

Reply:
Looks great! Yes, I'll need to do that before publishing.

Reply:
I've managed to simulate a GHPRB build and have a status reported with the correct coverage, so I'm happy with it from that end.

The only issue I'm having is getting one of the generated tests (testPluginActive) to pass. It would be useful to hear if this passes in your environment.

Usually this test indicate which dependency is causing the failure but in this case there is no nested exception so I don't have any lead on how to fix it.

Reply:
Test working! Appeared to be a caching issue with IntelliJ. 

Let me know if it's functional from your side for non-GHPRB jobs and if you're happy I'll merge.

Release will likely be tomorrow now.

Reply:
I am rolling out to my Jenkins, but they are heavily used :-). I need a maintenance window to restart Jenkins. I will let you know.

Reply:
I confirm that the latest code works on our Jenkins ;-)
![Screen Shot 2019-07-17 at 5 09 50 PM](https://user-images.githubusercontent.com/32844327/61412055-d2c23280-a8b5-11e9-8468-f489fcb8c4aa.png)

Reply:
wait.
it is failing when I merge to master. I need to check how to deal with that

Reply:
I change the pipeline step to
```
        stage('Testing...') {
            steps {
                ...
            }
            post {
                success {
                    // if we are in a PR
                    if (env.CHANGE_ID) {
                        publishCoverageGithub(filepath:'coverage.xml', coverageXmlType: 'cobertura', comparisonOption: [ value: 'optionFixedCoverage', fixedCoverage: '0.65' ], coverageRateType: 'Line')
                    }
                }
            }
        }
```

Reply:
here is the (new) working code:
```
        stage('Testing...') {
            steps {
                ...
            }
            post {
                success {
                    // if we are in a PR
                    script {
                        if (env.CHANGE_ID) {
                            publishCoverageGithub(filepath:'coverage.xml', coverageXmlType: 'cobertura', comparisonOption: [ value: 'optionFixedCoverage', fixedCoverage: '0.65' ], coverageRateType: 'Line')
                        }
                    }
                }
            }
        }
```

Let me know if we do that, or we "fail" silently if we are not in a PR...

Reply:
I also saw that error but it occurred when I wasn't passing in the GHPRB variables. I'm not familiar with CHANGE_URL as an env var I'm afraid. Which plugin does it come from?

Reply:
I've had a look at the wiki and this variable comes from the multi-branch project plugin. In short, this isn't going to work for a lot of use cases / standard build jobs.

Reply:
GIT_URL could work but the regex will need to be reworked to account for the different possible git url formats (git@github.com:my/repo.git, https://github.com/my/repo, ssh://git@github.com/my/repo, etc.)

Reply:
multi-branch project plugin IS Jenkinsfile AFAIK ;-)
So these variables (cf https://hudson.eclipse.org/webtools/env-vars.html/) are particularly useful for Jenkinsfile DSL

Reply:
Oh! in that case I think it's fine. As long as GHPRB is still supported (which it is per my testing), and pipeline is supported per your use case, then I'm fine with it 😄 

Are you happy for me to merge?

Reply:
I am happy with that.
And let's see if I/we receive feedback for improvement
I dont know where we can post also my example (for Jenkinsfile)

Reply:
I would put it in the README - please raise that as a separate PR as I can merge that quickly without building.